src/main/java/gregtech/integration/jei/recipe/GTRecipeWrapper.java

```diff
@@ -152,11 +151,20 @@ public void addFluidTooltip(int slotIndex, boolean input, Object ingredient, Lis
     public void drawInfo(@Nonnull Minecraft minecraft, int recipeWidth, int recipeHeight, int mouseX, int mouseY) {
         super.drawInfo(minecraft, recipeWidth, recipeHeight, mouseX, mouseY);
         int yPosition = recipeHeight - recipeMap.getPropertyListHeight(recipe);
-        if (!recipe.hasProperty(PrimitiveProperty.getInstance())) {
+
+        // Default entries
+        var properties = recipe.getPropertyTypes();
+        if (properties.isEmpty() || properties.stream().allMatch(RecipeProperty::showTotalEU)) {
             minecraft.fontRenderer.drawString(I18n.format("gregtech.recipe.total", Math.abs((long) recipe.getEUt()) * recipe.getDuration()), 0, yPosition, 0x111111);
+        } else yPosition -= LINE_HEIGHT;
+        if (properties.isEmpty() || properties.stream().allMatch(RecipeProperty::showEUt)) {
             minecraft.fontRenderer.drawString(I18n.format(recipe.getEUt() >= 0 ? "gregtech.recipe.eu" : "gregtech.recipe.eu_inverted", Math.abs(recipe.getEUt()), GTValues.VN[GTUtility.getTierByVoltage(recipe.getEUt())]), 0, yPosition += LINE_HEIGHT, 0x111111);
-        } else yPosition -= LINE_HEIGHT * 2;
-        minecraft.fontRenderer.drawString(I18n.format("gregtech.recipe.duration", TextFormattingUtil.formatNumbers(recipe.getDuration() / 20d)), 0, yPosition += LINE_HEIGHT, 0x111111);
+        } else yPosition -= LINE_HEIGHT;
+        if (properties.isEmpty() || properties.stream().allMatch(RecipeProperty::showDuration)) {
```

Comment:
Should these be `anyMatch` instead of `allMatch`? Because if the properties have showDuration and showEUt, wouldn't allMatch return false for because it didn't match showEUt?

Reply:
My thinking is that if a property has one of these set to false, do not show it even if other properties say to. Might be better to name them "hideEUt", "hideTotalEU", etc